### ISSUE TITLE
Module separation for Time4J and Time4A

### ISSUE THREAD

**Issue by moallemi** [MenoData/Time4J#898]
First of all thank you for your great work on creating and maintaining this library.

As a developer who wants to access other calendars than gregorian, I usually need only one more calendar (persian calendar) in my project. I want to ask you if it is possible for you to separate the calendars from the base artifact and make it like this:

```
compile    'net.time4j:time4j-base:5.5'
compile   'net.time4j:time4j-base-CALENDAR_NAME:5.5'
```
so If i need the persian calendar I can only include 
`compile      'net.time4j:time4j-base-persian-calendar:5.5'`
in my dependencies.

Looking forward to hearing from you.

**Comment by MenoData** [comment]
More modules automatically mean more maintenance effort in every single release. So a new module needs some special justification.

**When to use modules?** Just the fact that you only want to use some parts of Time4J is not enough to justify a module split IMHO. A better justification would be the need for a smaller library, but this size requirement is not really relevant on standard computers in a standard JVM (where disk space is cheap and available in huge amounts). Indeed, there are two other modules (SQL and XML) beyond the base-module but this is necessary not because of the size of the library but because of extra dependencies on SQL- or XML-packages of Java which are not always part of standard basic JVM-distributions.

**Android however, really asks for small libraries.** So theoretically, a module split would be a good idea. I have experimented in the past in that direction but did not find a good way to unify Java-9-modules with Android due to very different mechanisms of resource loading (Time4J/A uses many resources for internationalization). So how to get Time4A small? Here I suggest to apply Proguard in order to wipe out all unused parts of Time4A out of the compiled apk-file.  This appears to be apparently more useful than using modules on Android platforms.

**Historical note:** Before October 2018 (v4.0-2018g), there was a problem because most calendar classes used the same serialization stub class (common SPX-class) and prevented Proguard from removing unused calendars. This problem has been solved. Now every calendar has its own serialization stub, see this [commit](https://github.com/MenoData/Time4A/commit/966664c4f53babea9e4eae23ae5996d160a00f79#diff-0f48a5a4c28d77b2add5ee1d64c65194).
